presto-main/src/main/java/com/facebook/presto/operator/PagesIndex.java

```diff
@@ -190,6 +206,9 @@ public void addPage(Page page)
             valueAddresses.add(sliceAddress);
         }
 
+        if (eagerCompact) {
+            compact();
```

Comment:
This results in `calculateEstimatedSize` being called twice.

Reply:
Nice catch! Thank you !!